relay-server/src/actors/processor.rs

```diff
@@ -2741,70 +2741,80 @@ impl EnvelopeProcessorService {
-                    }
-                }
-
-                relay_log::trace!("inserting metric buckets into project cache");
-                self.inner
-                    .project_cache
-                    .send(MergeBuckets::new(public_key, buckets));
-            } else if item.ty() == &ItemType::MetricBuckets {
-                match serde_json::from_slice::<Vec<Bucket>>(&payload) {
-                    Ok(mut buckets) => {
-                        for bucket in &mut buckets {
-                            clock_drift_processor.process_timestamp(&mut bucket.timestamp);
-                        }
-
-                        relay_log::trace!("merging metric buckets into project cache");
-                        self.inner
-                            .project_cache
-                            .send(MergeBuckets::new(public_key, buckets));
+                        );
+                        None
                     }
+                })
+                .collect(),
+            ItemType::MetricBuckets => {
+                match serde_json::from_slice::<Vec<Bucket>>(&item.payload()) {
+                    Ok(buckets) => buckets,
                     Err(error) => {
                         relay_log::debug!(
                             error = &error as &dyn Error,
                             "failed to parse metric bucket",
                         );
                         metric!(counter(RelayCounters::MetricBucketsParsingFailed) += 1);
+                        vec![]
                     }
                 }
-            } else {
+            }
+            _ => {
                 relay_log::error!(
                     "invalid item of type {} passed to ProcessMetrics",
                     item.ty()
                 );
+                vec![]
             }
         }
     }
 
+    fn handle_process_metrics(&self, message: ProcessMetrics) {
+        let ProcessMetrics {
+            items,
+            project_key: public_key,
+            start_time,
+            sent_at,
+        } = message;
+
+        let received = relay_common::time::instant_to_date_time(start_time);
+        let received_timestamp = UnixTimestamp::from_secs(received.timestamp() as u64);
+
+        let mut buckets: Vec<Bucket> = items
+            .iter()
+            .flat_map(|item| Self::extract_metric_buckets_from_item(item, received_timestamp))
+            .collect();
+
+        if buckets.is_empty() {
+            relay_log::trace!("no metric buckets could be extracted from items");
+            return;
+        };
+
+        let clock_drift_processor =
+            ClockDriftProcessor::new(sent_at, received).at_least(MINIMUM_CLOCK_DRIFT);
+
+        for bucket in &mut buckets {
+            clock_drift_processor.process_timestamp(&mut bucket.timestamp);
+        }
+
+        relay_log::trace!("inserting metric buckets into project cache");
+        self.inner
+            .project_cache
+            .send(MergeBuckets::new(public_key, buckets));
```

Comment:
The Mergebuckets message is now sent only once, but with all the buckets, is there any implications of this that i am not aware of?